apprunner-gradle-plugin/src/main/java/org/projectnessie/quarkus/gradle/QuarkusAppExtension.java

```diff
@@ -50,6 +50,8 @@ public QuarkusAppExtension(Project project) {
         project.getObjects().property(String.class).convention("quarkus.http.test-url");
     httpListenPortProperty =
         project.getObjects().property(String.class).convention("quarkus.http.test-port");
+    // set the port as the property, to allow clients of the extension to get the port.
+    System.setProperty("quarkus.http.test-port", httpListenPortProperty.get());
```

Comment:
That'll definitely not work. You're exposing the value of the `httpListenPortProperty` which is the _name_ of the property.

Using system properties is also not good, because Gradle's meant to run things in parallel - so this will break, if two Gradle projects use this plugin.

Reply:
Good point. 
I see that it is solved in the non-scala modules by using:
```
 <artifactId>maven-failsafe-plugin</artifactId>
        <executions>
          <execution>
            <goals>
              <goal>integration-test</goal>
              <goal>verify</goal>
            </goals>
          </execution>
        </executions>
        <configuration>
          <systemPropertyVariables>
            <quarkus.http.test-port>${quarkus.http.test-port}</quarkus.http.test-port>
          </systemPropertyVariables>
          <additionalClasspathElements>${project.build.directory}/test-sources</additionalClasspathElements>
          <failIfNoTests>true</failIfNoTests>
        </configuration>
```
However, the Gatling module does not use the failsafe plugin. 
Maybe we should try to do something similar to:
```
 <artifactId>nessie-apprunner-maven-plugin</artifactId>
        <configuration>
          <skip>${skipITs}</skip>
          <systemProperties>
            <!-- Use INMEMORY version store for testing -->
            <nessie.version.store.type>INMEMORY</nessie.version.store.type>
          </systemProperties>
          <environment>
            <HTTP_ACCESS_LOG_LEVEL>${test.log.level}</HTTP_ACCESS_LOG_LEVEL>
          </environment>
        </configuration>
```
and the injection:
```
<HTTP_ACCESS_LOG_LEVEL>${test.log.level}</HTTP_ACCESS_LOG_LEVEL>
```
for the 
```
<systemPropertyVariables>
    <quarkus.http.test-port>${quarkus.http.test-port}</quarkus.http.test-port>
</systemPropertyVariables>
```

Reply:
**update**:
I've tested adding:
```
<quarkus_http_test_port>${quarkus.http.test-port}</quarkus_http_test_port>
```
to the `<artifactId>nessie-apprunner-maven-plugin</artifactId>` for `<systemProperties>` and `<systemProperties>` but the setting is not properly injected.

Reply:
Not sure I understand what you're trying to do when adding that to the apprunner plugin, but that `${quarkus.http.test-port}` needs to be used in the `gatling-maven-plugin`, likely using `<jvmArg>-Dquarkus.http.test-port=${quarkus.http.test-port}</jvmArg>` (or whatever the name of the properties are).

Reply:
good point changed in the: https://github.com/projectnessie/nessie/pull/3647 PR.
Closing this one